src/builtins.rs

```diff
@@ -59,12 +61,88 @@ lazy_static! {
                     }
                 }
             },
+        ),
+        (
+            "MUX",
+            BuiltIn {
+                decl: "FUNCTION MUX<U: ANY> : U
+                VAR_INPUT
+                    K : DINT;
+                    args : {sized} U...;
+                END_VAR
+                END_FUNCTION
+                ",
+                transformation: |it| it,
+                code: |generator, params, location| {
+                    //Generate an access from the first param
+                    if let &[k, ..] = params {
+                        let k = generator.generate_expression(k)?;
+                        let pou = generator.index.find_pou("MUX").expect("MUX exists as builtin");
+                        //Generate a pointer for the rest of the params
+                        let params = generator.generate_variadic_arguments_list(pou, &params[1..])?;
+                        //First access is into the array
+                        let ptr = generator.llvm.load_array_element(params[1].into_pointer_value(),&[generator.llvm.context.i32_type().const_zero(), k.into_int_value()],"")?;
+                        Ok(generator.llvm.builder.build_load(ptr, ""))
```

Comment:
we can replace this with llvm-select instruction

https://llvm.org/docs/LangRef.html#select-instruction

it probably also makes it easier to optimize if a literal is used

Reply:
For the SEL this would work yes, but I don't think it works for mux since it requires a boolean

Reply:
oh sry you're right, I probably meant to put this on the SEL :smile: